mygame/app/scenes/team_builder.rb

```diff
@@ -0,0 +1,109 @@
+module Scenes
+  class TeamBuilder
+    def initialize(args)
+      @font = build_pokemini_font
+      @fatnumbers_font = build_pokemini_fatnumbers_font
+
+      @team = args.state.team_builder.team ||= [nil, nil, nil]
+
+      @slots = @team.size.times.map do |i|
+        button(x: 42 - i * 20, y: 30, w: 18, h: 18, bgcolor: Palette::WINDOW_BG_COLOR)
```

Comment:
Here's the culprit! you're iterating from right to left :P

Reply:
I'm learning Japanese, that's why :P